### PR TITLE
Issue #SH-SB-21311 feat: download activity progress as CSV

### PR DESCRIPTION
**Pull request by balakrishna10**

### CODE REVIEW

**Review by swayangjit — COMMENTED** [review]
@balakrishna10   Please add testcases and remove all duplicated code suggested by Sonar

File: src/app/my-groups/activity-details/activity-details.page.ts [hunk 1]
```diff
@@ -237,4 +249,123 @@ export class ActivityDetailsPage implements OnInit, OnDestroy {
...
+                }
+                resolve(undefined);
+              });
+          }
+        }, this.appName, this.commonUtilService.translateMessage('FILE_MANAGER'), 'FILE_MANAGER_PERMISSION_DESCRIPTION', PageId.PROFILE, true
+      );
+      await confirm.present();
+    });
+  }
+
```
Comment (swayangjit) [inline src/app/my-groups/activity-details/activity-details.page.ts:361]: @balakrishna10 these line of code is almost in 3 4 page . Can you move it  to a separate service or permission service

**Comment by swayangjit** [comment]
Remove 2 code smells suggested by Sonar

File: src/app/my-groups/activity-details/activity-details.page.ts (hunk 1 above)
Comment (balakrishna10) [reply src/app/my-groups/activity-details/activity-details.page.ts:361]: Can I do it in another PR ? that will need more changes in multiple files. I can work on it next week.

**Review by AmiableAnil — APPROVED** [review]

**Pull request merged**